osu.Game.Rulesets.Mania/Beatmaps/Patterns/Legacy/DistanceObjectPatternGenerator.cs

```diff
@@ -483,9 +483,12 @@ private List<IList<HitSampleInfo>> nodeSamplesAt(double time)
             if (!(HitObject is IHasPathWithRepeats curveData))
                 return null;
 
-            double segmentTime = (EndTime - HitObject.StartTime) / spanCount;
-
-            int index = (int)(segmentTime == 0 ? 0 : (time - HitObject.StartTime) / segmentTime);
+            // mathematically speaking this could be done by calculating (time - HitObject.StartTime) / SegmentDuration
+            // however, floating-point operations can introduce inaccuracies - therefore resort to iterated addition
+            // (all callers use this method to calculate repeat point times, so this way is consistent and deterministic)
+            int index = 0;
+            for (double nodeTime = HitObject.StartTime; nodeTime < time; nodeTime += SegmentDuration)
```

Comment:
How do you feel about `Math.Round`ing this? Given that `sliderDuration = n * segmentDuration`, `sliderDuration / segmentDuration` should give us a value that is approximately equal to `n` and `Math.Round` handles sanitisation.

Seems to pass the test cases too.

Reply:
I was a little worried that the round could round things that it shouldn't but maybe it's a bit paranoid.

A bit of an awkward situation, this. Ideally I'd say `nodeSamplesAt` should just take in an integer index but the surrounding code is structured such that doing that is not straightforward

Reply:
I've changed my mind on this and pushed in the rounding variant. This is a private function and one that shouldn't see much traffic going forward, so I now feel that being worried that someone will call this method with `time` somewhere between repeats is unreasonable.